src/System.Net.Http/src/System/Net/Http/Managed/HttpEnvironmentProxy.cs

```diff
@@ -0,0 +1,261 @@
+        }
+
+        private HttpEnvironmentProxy(Uri httpProxy, Uri httpsProxy, string bypassList)
+        {
+            NetworkCredential httpCred = null;
+            NetworkCredential httpsCred = null;
+            bool sameAuth = false;
+
+            if (httpProxy != null)
+            {
+                _httpProxyUri = httpProxy;
+                httpCred = GetCredentialsFromString(_httpProxyUri.UserInfo);
+            }
+            if (httpsProxy != null)
+            {
+                _httpsProxyUri = httpsProxy;
+                httpsCred = GetCredentialsFromString(_httpsProxyUri.UserInfo);
+                if ((_httpProxyUri != null) && (_httpProxyUri.UserInfo == _httpsProxyUri.UserInfo))
+                {
+                    sameAuth = true;
+                }
+            }
+            if (httpCred != null || httpsCred != null)
+            {
+                // If only one protocol is set or both protocols have same auth,
+                // use standard credential object
+                if (sameAuth || httpCred == null || httpsCred == null)
+                {
+                    _credentials = httpCred ?? httpsCred;
+                }
+                else
+                {
+                    // use wrapper so we can decide later based on uri
+                    _credentials = new HttpEnvironmentProxyCredentials(httpCred, httpsCred);
+                }
+            }
+
+            if (!string.IsNullOrWhiteSpace(bypassList))
+            {
+                string[] list = bypassList.Split(',');
+                List<string> tmpList = new List<string>();
+
+                foreach (string value in list)
+                {
+                    string tmp = value.Trim();
+                    if (tmp.Length > 0)
+                    {
+                        tmpList.Add(tmp);
+                    }
+                }
+                if (tmpList.Count > 0)
+                {
+                    _bypass = tmpList.ToArray();
+                }
+            }
+        }
+
+        /// <summary>
+        /// This function will evaluate given string and it will try to convert
+        /// it to Uri object. The string could contain URI fragment, IP address and  port
+        /// tuple or just IP address or name. It will return null if parsing fails.
+        /// </summary>
+        private static Uri GetUriFromString(string value)
+        {
+            if (string.IsNullOrEmpty(value))
+            {
+                return null;
+            }
+
+            if (!value.Contains("://"))
+            {
+                value = "http://" + value;
+            }
+
+
+
+            Uri uri = null;
+            try
+            {
+                uri = new Uri(value);
+            }
+            catch
+            {
+                return null;
+            }
+
+            if (uri == null || (uri.Scheme != Uri.UriSchemeHttp && uri.Scheme != Uri.UriSchemeHttps))
```

Comment:
This if block can be inside the try statement, so you don't have to declare Uri outside the try block..